libraries/AP_EFI/AP_EFI_MAV.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_EFI_config.h"
+
+#if AP_EFI_MAV_ENABLED
+
+#include "AP_EFI_MAV.h"
+#include <AP_Math/AP_Math.h>
+
+//Called from frontend to update with the readings received by handler
+void AP_EFI_MAV::update()
+{
+    if (receivedNewData) {
+    	copy_to_frontend();
+    	receivedNewData = false;
+    }
+}
+
+//Decode MavLink message
+void AP_EFI_MAV::handle_EFI_message(const mavlink_message_t &msg)
+{
+    mavlink_efi_status_t state;
+    mavlink_msg_efi_status_decode(&msg, &state);
+
+    internal_state.ecu_index = state.ecu_index;
+    //internal_state.??? = state.rpm;
```

Comment:
Check my PR for mappings for this.

https://github.com/ArduPilot/ardupilot/commit/8d52efbf8c44353012455d66926a8d99cfa452a0#diff-fd3b580d602da071426c9e0387b1cc0283213a5d883b00a5ae7cf89467ea4e09R15